src/server/pfs/drive/driver.go

```diff
@@ -593,6 +625,11 @@ func (d *driver) AddShard(shard uint64) error {
 				if err := d.insertDiffInfo(diffInfo); err != nil {
 					return err
 				}
+				if diffInfo.Finished == nil {
+					if _, ok := d.commitConds[diffInfo.Diff.Commit.ID]; !ok {
+						d.commitConds[diffInfo.Diff.Commit.ID] = sync.NewCond(&d.lock)
+					}
+				}
```

Comment:
No need for this, `DiffInfo`s only get serialized when the commit is finished. So we can assume that the commits we receive in `AddShard` are all finished.
Why don't we make that an assert in this function so the invariant is obvious?